pkg/bpf/ads/sock_ops.go

```diff
@@ -180,18 +177,37 @@ func (sc *BpfSockOps) Load() error {
 }
 
 func (sc *BpfSockOps) Attach() error {
+	var err error
 	cgopt := link.CgroupOptions{
 		Path:    sc.Info.Cgroup2Path,
 		Attach:  sc.Info.AttachType,
 		Program: sc.KmeshSockopsObjects.SockopsProg,
 	}
 
-	lk, err := link.AttachCgroup(cgopt)
-	if err != nil {
+	// pin bpf_link and bpf_tail_call map
+	// pin bpf_link, after restart, update prog in bpf_link
+	// tail_call map cannot pin in SetMapPinType->LoadAndAssign, we pin them independent
+	mapPinPath := filepath.Join(sc.Info.BpfFsPath, "sockops_tail_call_map")
+	progPinPath := filepath.Join(sc.Info.BpfFsPath, "sockops_link")
+	if restart.GetStartType() == restart.Restart {
+		if sc.Link, err = utils.BpfProgUpdate(progPinPath, cgopt); err != nil {
+			return err
+		}
+		if err = utils.BpfMapDeleteByPinPath(mapPinPath); err != nil {
```

Comment:
```suggestion
		os.Remove(mapPinPath)
```
The return value does not need to be checked for error tolerance.